docs/source/guides/scripting/customwavecreation.rst

```diff
@@ -0,0 +1,479 @@
+- ``FD_spawnPosition(<X,Y,Z>)`` and ``FD_spawnAngles(<Y,A,P>)``
+    - Coordinates and Angles at which the Dropship animation drops you off at ``spawnPosition`` and angle of the dropship ``spawnAngles``.
+    - They are both required or the game will crash.
+
+.. note::
+    The image below explains roughly what you need to lookout for when putting your ``FD_spawnPosition`` on the map. As script coordinates is ``FD_spawnPosition``
+|DropshipAnimation|
+
+
+Logic Events
+############
+
+``wave#.append(CreateSmokeEvent(<X,Y,Z>,#inseconds,index++/0))``
+    - Empty Template: ``wave.append(CreateSmokeEvent(<>,,))``
+    - Ex: ``wave1.append(CreateSmokeEvent(< 4060,771,1004 >,90,index++))``
+
+Enemy Events
+############
+
+Normal AI
+^^^^^^^^^
+
+Titans
+^^^^^^
+
+Elite Titans
+^^^^^^^^^^^^
+.. Warning ::
+    The Elite Titans are from the `Experimental`_ branch! Use that branch if you want to use Elite Titans!
+
+.. Note ::
+    Special Titans ``Arc, Nuke, Mortar`` **cannot be elites**
+
+To make a Titan an elite, you must add an extra parameter at the end of the create event to set it to an elite:
+
+- ``FD_TitanType.TITAN_ELITE``
+
+And a serverside variable must also be enabled for said elite to spawn, without it, the elite will not spawn in even with the added parameter:
+
+- ``ns_fd_allow_elite_titans 1``
+
+An example of this would be: 
+
+- ``wave5.append(CreateLegionTitanEvent(<4779,-2194,-53>,<0,-170,0>,index++,1,"",FD_TitanType.TITAN_ELITE))``
+
+Routes
+######
+
+.. dropdown:: Forwardbase Kodai
+    - hillRouteClose
+    - 
+    
+Finding Coordinates To Use For Events
+#####################################
+
+The easiest way to start is to start a Private Match and make sure match settings for ``Time Limit`` is set to a higher value.
+
+Once in game enable ``sv_cheats 1`` and ``cl_showpos 1``. Also use ``noclip`` to get around the map to find your in game coordinates
+
+Once ``cl_showpos 1`` is enabled, in the top left a set of coordinates for both ``X,Y,Z`` and ``Y,A,P`` show up, these coordinates is what you input into the events to spawn Enemy AI.
+
+|In-GameCoordinates|
+
+Troubleshooting
+###############
+
+[SERVER] Index "#" is beyond array size of #
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+What this typically means is when you don't end a wave.append with ``0`` instead of ``index++`` at the end of the WaveEvent array.
+
+Instead of ending with index++:
+
+.. code-block::
+
+    wave1.append(CreateDroppodGruntEvent(< 4120,800,1004 >,"",index++))
+    wave1.append(CreateDroppodGruntEvent(< 4150,730,1004 >,"",index++))
+    waveEvents.append(wave1)
+    index=1
+    array<WaveEvent> wave2
+
+It needs to end with 0:
+
+.. code-block:: 
+
+    wave1.append(CreateDroppodGruntEvent(< 4120,800,1004 >,"",index++))
+    wave1.append(CreateDroppodGruntEvent(< 4150,730,1004 >,"",0))
+    waveEvents.append(wave1)
+    index=1
+    array<WaveEvent> wave2
+
+COMPILE ERROR expected expression; found "<-"
```

Comment:
they should just read the squirrel docs might want some feedback from someone else on this